example/single/single.cpp

```diff
@@ -117,19 +117,19 @@ options parse_options(int argc, char** argv) {
 
     char** arg = argv+1;
     while (*arg) {
-        if (auto dt = parse_opt<double>(arg, 'd', "dt")) {
+        if (auto dt = parse<double>(arg, 'd', "dt")) {
             opt.dt = dt.value();
         }
-        else if (auto t_end = parse_opt<double>(arg, 't', "t-end")) {
+        else if (auto t_end = parse<double>(arg, 't', "t-end")) {
             opt.t_end = t_end.value();
         }
-        else if (auto weight = parse_opt<float>(arg, 'w', "weight")) {
+        else if (auto weight = parse<float>(arg, 'w', "weight")) {
             opt.syn_weight = weight.value();
         }
-        else if (auto swc = parse_opt<std::string>(arg, 'm', "morphology")) {
+        else if (auto swc = parse<std::string>(arg, 'm', "morphology")) {
             opt.swc_file = swc.value();
         }
-        else if (auto nseg = parse_opt<unsigned>(arg, 'n', "cv-per-branch")) {
+        else if (auto nseg = parse<unsigned>(arg, 'n', "cv-per-branch")) {
             opt.policy = arb::cv_policy_fixed_per_branch(nseg.value(), arb::cv_policy_flag::single_root_cv);
         }
```

Comment:
While this is the most direct adaption, an alternative would be to use the `<<` operator, for something like the following:
```
    auto set_policy = [&opt](int n) {
        opt.policy = arb::cv_policy_fixed_per_branch(n, arb::cv_policy_flag::single_root_cv);
    };

    char** arg = argv+1;
    while (*arg) {
        bool ok =
            opt.dt << parse<double>(arg, 'd', "dt") ||
            opt.t_end << parse<double>(arg, 't', "t-end") ||
            opt.syn_weight << parse<float>(arg, 'w', "weight") ||
            opt.swc_file << parse<std::string>(arg, 'm', "morphology") ||
            set_policy << parse<unsigned>(arg, 'n', "cv-per-branch");

        if (!ok) […] // usage error etc.
   }
```
The code you have though is somewhat less magical looking, and may be preferable!

Reply:
While I like this implementation (especially how compact it is), I think the original code is slightly more readable which may be preferable in the example code. However, if you disagree with that I have no problem using this version :)